packages/components/src/internal/components/files/FileTree.spec.tsx

```diff
@@ -38,9 +45,190 @@ describe('FileTree', () => {
+        wrapper.unmount();
+    });
+
+    test('file node, with checkbox', () => {
+        const wrapper = mount(<Header {...DEFAULT_PROPS} handleCheckbox={jest.fn} />);
+        validate(wrapper, false, false, true);
+        wrapper.unmount();
+    });
+
+    test('directory node', () => {
+        const wrapper = mount(
+            <Header {...DEFAULT_PROPS} node={{ id: 'test-id', active: false, children: [], name: 'test name' }} />
+        );
+        validate(wrapper, false, false, false, true, true);
+        expect(wrapper.find('.filetree-leaf-node')).toHaveLength(0);
+        wrapper.unmount();
+    });
+
+    test('showNodeIcon false', () => {
+        const wrapper = mount(<Header {...DEFAULT_PROPS} showNodeIcon={false} />);
+        validate(wrapper, false, false, false, false);
+        wrapper.unmount();
+    });
+
+    test('active and not allowMultiSelect', () => {
+        const wrapper = mount(
+            <Header {...DEFAULT_PROPS} node={{ id: 'test-id', active: true, children: undefined, name: 'test name' }} />
+        );
+        validate(wrapper);
+        expect(wrapper.find('.active')).toHaveLength(1);
+        expect(wrapper.find('.lk-text-theme-dark')).toHaveLength(1);
+        wrapper.unmount();
+    });
+
+    test('active and allowMultiSelect', () => {
+        const wrapper = mount(
+            <Header
+                {...DEFAULT_PROPS}
+                node={{ id: 'test-id', active: true, children: undefined, name: 'test name' }}
+                allowMultiSelect
+            />
+        );
+        validate(wrapper);
+        expect(wrapper.find('.active')).toHaveLength(1);
+        expect(wrapper.find('.lk-text-theme-dark')).toHaveLength(0);
+        wrapper.unmount();
+    });
+
+    test('empty node', () => {
+        const wrapper = mount(<Header {...DEFAULT_PROPS} node={{ id: 'test|' + EMPTY_FILE_NAME }} />);
+        validate(wrapper, true);
+        wrapper.unmount();
+    });
+
+    test('loading node', () => {
+        const wrapper = mount(<Header {...DEFAULT_PROPS} node={{ id: 'test|' + LOADING_FILE_NAME }} />);
+        validate(wrapper, true, true);
+        wrapper.unmount();
+    });
+
+    test('customStyles and selected', () => {
+        const title = { fontWeight: 'normal' };
+        const customTitle = { fontWeight: 'bold' };
+        let wrapper = mount(
+            <Header
+                {...DEFAULT_PROPS}
+                node={{
+                    id: 'test-id',
+                    active: false,
+                    children: undefined,
+                    name: 'test name',
+                    selected: false,
+                }}
+                style={{ base: {}, title }}
+                customStyles={{ header: { title: customTitle } }}
+            />
+        );
+        validate(wrapper);
+        expect(wrapper.find('.filetree-resource-row').prop('style')).toStrictEqual(title);
+        wrapper.unmount();
+
+        wrapper = mount(
```

Comment:
Any reason not to split this into a separate test?  It's testing a slightly different case that doesn't depend on the previous test that I can see.

Reply:
Good point. Spit into two test cases.